cfme/utils/appliance/__init__.py

```diff
@@ -2741,6 +2741,19 @@ def disable_migration_ui(self):
         if self.advanced_settings.get('product', {}).get('transformation'):
             self. _switch_migration_ui(False)
 
+    def set_public_images(self, enable, provider_type):
+        provider = 'ec2'
```

Comment:
**Optional**

I would consider defining this method on `CloudProvider`, and using `self.appliance.advanced_settings`.  I would also consider not checking the current setting here, and always calling `update_advanced_settings`.

Whether or not the field is currently enabled/disabled seems like it would be far more useful modeled as a property of the appliance (Sentaku + ViaREST?)

**Required**
Use provider attributes to get 'ec2' and 'azure' strings, instead of checking passed provider_type. Take `provider` object instead of type string.

Reply:
Well I didn't want to define this method on `CloudProvider` because it's set across whole appliance and when I instantiate a `CloudProvider` it also impacts all the other instances of it...